pandas/io/common.py

```diff
@@ -174,7 +175,7 @@ def _expand_user(filepath_or_buffer: str | BaseBufferT) -> str | BaseBufferT:
     return filepath_or_buffer
 
 
-def validate_header_arg(header: object) -> None:
+def validate_header_arg(header: int | Sequence[int] | None) -> None:
```

Comment:
The function accepts everything and only throws an error in case a `bool` is provided. I also see a point in typing it for the expected types to find errors in places that call this function.

Reply:
Hm this was a mistake, did not want to change that. Thanks for mentioning. 

Adding to that: I think we should redesign that. We currently have to places where we validate the header...